README.md

```diff
@@ -186,3 +186,161 @@ If none package suits scope of new changes, we need to create new package by cre
 Webhooks are enabled to trigger releases on travis from comment on merged PR. If you are member of group responsible for releases you can add new commnent to merged PR `Release minor`, `Release bugfix` or `Release` in order to trigger new release.
 
 You can also draft a release by adding label `release` or `release minor` and once this PR is merged new release will be triggered.
+
+## Typescript guidelines
+
+Typescript build has been finally introduced to the repository. All modules will be eventually refactored into typescript. When refactoring, or creating new modules, please follow these instructions when creating types:
+
+### React components
+
+1. All component props must be an interface and must be exported. This is required for consuming apps to use features like `omit` or `extends` when developing with typescript. A nice example is generating JSX from array structure, but for some reseason extension/exclustion of the interface is required.
+
+```TS
+
+import React from 'react';
+import { AlertProps, AlertVariant } from '@patternfly/react-core';
+
+/**
+ * Custom component wrapper from FEC
+ * It has a description prop which is mapped to the PF Alert children prop
+ * */
+interface AlertWrapper extends AlertProps {
+  description?: string
+}
+
+/**
+ * Local type which builds on FEC defined type
+*/
+interface LocalAlertType extends AlertWrapper {
+  id: string
+}
+
+const CustomComponent: React.ComponentType = ({ data }) => {
+    const alertsStructure: LocalAlertType[] = data.map(({ uuid, status, title, message }) => ({
+        id: uuid,
+        variant: status === 500 ? AlertVariant.danger : AlertVariant.info,
+        title,
+        description: message
+    }));
+
+    return alertsStructure.map(({ id, ...props }) => <AlertWrapper key={id} {...props} />);
+};
+
+```
+
+2. When creating a wrapper around PF components, always extend original components props. This will ensure that all original props are accepted by your component wrapper. If you have to exclude some original prop, use `Omit` generic. Exclusion might be required because the wrapper is setting a prop by default.
```

Comment:
God damn, I wanted to go with the types as they are giving us more flexibility. But upon looking at PF they straight up exort `interface` and that's it. Looks like we don't have any other chance (unless we want to have a mix of types and interfaces)

Reply:
Yup. That is why I would go with interfaces. As we build on top PF most of the time, it is probably for the best to keep it same.